instrumentation/spring/starters/spring-boot-starter/build.gradle.kts

```diff
@@ -12,6 +12,7 @@ dependencies {
   api("org.springframework.boot:spring-boot-starter:$springBootVersion")
   api("org.springframework.boot:spring-boot-starter-aop:$springBootVersion")
   api(project(":instrumentation:spring:spring-boot-autoconfigure"))
+  implementation(project(":instrumentation:resources:library"))
```

Comment:
why did you choose to add this as `implementation` when all the other dependencies are `api`?
secondly these resource providers use the SPI that the sdk autoconfigure module uses, but spring boot starter does not uses the sdk autoconfigure. Even if you add them do they actually do anything?

Reply:
I would see more the new dependency included in `spring:spring-boot-autoconfigure`.

Reply:
1. implementation is more conservative (doesn't allow access to transitive dependencies), so I use that by default
2. .. does not use autoconfigure ... -> I've been wondering the same - is there there a good reason for that?
3. will it work? Yes, there's a [guard that checks if the class exists](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/resources/OtelResourceAutoConfiguration.java#L48)

Reply:
@zeitlinger Could you develop one feature for OTel Spring autoconfiguration with this dependency and with one test?

Reply:
> I would see more the new dependency included in `spring:spring-boot-autoconfigure`.

autoconfigure modules should be as thin as possible, and starters add what's usually needed.

Reply:
The resource library is [compileOnly in Spring autoconfiguration](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/9e208c85740dfa5035c29ba84a419ff94b9ee519/instrumentation/spring/spring-boot-autoconfigure/build.gradle.kts#L49). Do we want to include it by default?